.github/workflows/pull-request-commit.yml

```diff
@@ -14,9 +14,109 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  webpack-analyzer:
+    runs-on: ubuntu-22.04
+    if: ${{ github.event_name == 'pull_request' }}
+    permissions:
+      pull-requests: write
+    steps:
+      - name: ⬇️ Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+
+      - name: 🔧 Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: yarn
+
+      - name: ⚙️ Install Dependencies
+        run: yarn install
+
+      - name: 🔦 Generate stats file for PR
+        run: |
+          yarn generate-webpack-stats-file
+          mv stats.json ../stats-new.json
+
+      - name: Get the base commit
+        id: base-commit
+        run: echo base-commit=$(git show-ref -s $(git describe --abbrev=0 --tags)) >> "$GITHUB_OUTPUT"
```

Comment:
Thinking about this again. This uses the latest _tag_ as the base commit to compare to. I think this makes the most sense, but maybe it would be better to just use the latest commit to main. If we merge in 4 changes that reduce the bundle size by 300kb, then create another commit that increases the size by 10KB, we'd still see a reduction of 290kb even though we actually increased the size in this PR.

Reply:
Changing it to the latest main commit for now, but if we do end up wanting to use a tag, this is what I used before: `        run: echo base-commit=$(git show-ref -s $(git describe --abbrev=0 --tags)) >> "$GITHUB_OUTPUT"`.

Or if we have a better thought on how to do it, totally can!

Reply:
Okay, I changed this to use the latest commit from the PR's base branch. Because we want to check _only_ the changes made in the PR, we will merge in the base commit to the PR's branch before generating the stats file. See comment below.